packages/care-package/import-vapp/src/BuildActions.ts

```diff
@@ -0,0 +1,116 @@
+        console.log('configuration: ' +  JSON.stringify(configuration));
+
+        const catalogId = await this.getOrCreateCatalogId(configuration);
+
+        await this.uploadVAppTemplate(catalogId, configuration);
+    }
+
+    private async getOrCreateCatalogId(configuration: Configuration) {
+        const catalogQuery = '/api/query?type=catalog&filter=name==' + configuration.catalog;
+        console.log('catalog_query: ' + catalogQuery);
+
+        const catalogs = await this.legacyApiClient.get<QueryResultRecordsType>(catalogQuery);
+        console.log(catalogs);
+
+        if (catalogs.total === 0) {
+            console.log('No care package catalog available. Creating...');
+            const adminCatalogType: AdminCatalogType = {
+                name: configuration.catalog,
+                description: 'Catalog created by the care package.',
+                isPublished: true
+            };
+
+            const catalogPost = '/api/admin/org/87ae2b63-da4e-4602-af80-0f722e1a1394/catalogs';
+            const catalog = await this.legacyApiClient.post<AdminCatalogType>(catalogPost, adminCatalogType);
+            const lastSlashPosition = catalog.id.lastIndexOf(':');
+            return catalog.id.substring(lastSlashPosition + 1);
+        } else {
+            const lastSlashPosition = catalogs.record[0].href.lastIndexOf('/');
+            return catalogs.record[0].href.substring(lastSlashPosition + 1);
+        }
+    }
+
+    private async uploadVAppTemplate(catalogId: string, configuration: Configuration) {
+        const catalogUploadUrl = '/api/catalog/' + catalogId + '/action/upload';
+
+        console.log('catalogUploadUrl ' + catalogUploadUrl);
+
+        const uploadVAppTemplate: UploadVAppTemplateParamsType = {
+            name: configuration.vAppTemplateName,
+            sourceHref: configuration.vAppTemplateURL,
+            description: configuration.vAppTemplateURL
+        };
+        console.log('uploadVAppTemplate: ' +  JSON.stringify(uploadVAppTemplate, null, 2));
+
+        const catalogItem = await this.legacyApiClient.post<CatalogItemType>(catalogUploadUrl, uploadVAppTemplate);
+        console.log('catalogItem: ' +  JSON.stringify(catalogItem, null, 2));
+    }
+
+    // private async instantiateVAppFromCatalogTemplate(catalogId: string, configuration: Configuration) {
+    //     const instantiateVAppTemplateUrl = '/api/vdc/a61bced4-168f-4944-9b1d-9b667c34dd62/action/instantiateVAppTemplate';
+    //
+    //     console.log('instantiateVAppTemplateUrl ' + instantiateVAppTemplateUrl);
+    //
+    //     const uploadVAppTemplate: InstantiateVAppTemplateParamsType = {
+    //         name: configuration.vAppTemplateName,
+    //         instantiationParams: {
+    //
+    //         },
+    //
+    //         isSourceDelete: false
+    //     };
+    //
+    // }
+
+    private init(apiConfig: CloudDirectorConfig) {
+        const providerOrgApiConfig = apiConfig.actAs('87ae2b63-da4e-4602-af80-0f722e1a1394');
```

Comment:
Should this be a configuration?

Reply:
[same as this previous one](https://github.com/vmware/vcd-ext-sdk/pull/227#discussion_r700263578)